app/Http/Controllers/ProjectController.php

```diff
@@ -5,18 +5,49 @@
 use App\Http\Requests\CreateProjectRequest;
 use App\Http\Requests\DeleteProjectRequest;
 use App\Project;
-use App\ProjectFollower;
-use App\File;
+use Illuminate\Pagination\LengthAwarePaginator;
+use Illuminate\Pagination\Paginator;
 use Illuminate\Support\Facades\Auth;
+use Illuminate\Support\Facades\Request;
 use Illuminate\Support\Facades\Session;
-use DB;
 
 class ProjectController extends Controller
 {
-	public function listProjects()
-	{
-		return view('project.list', ['projects' => Project::all()]);
-	}
+    /**
+     * The view for /projects. Lists all the projects taking into
+     * account the active page and the sorting of the projects.
+     *
+     * @return mixed
+     */
+    public function listProjects()
+    {
+        $allProjects = Project::all();
+        $sorting = Request::has('sort');
+
+        if ($sorting) {
+            $allProjects = $allProjects->sortBy(Request::get('sort'));
```

Comment:
@brec9824 Sorting happens here, it uses the built in Model's method of sorting so it's beyond our control.